### PR TITLE
Fixed lr too high for gradient accumulation fixed #3023

### LINKED ISSUE #3023: Gradient Accumulation causes lower lr to be same as non-gradient accumulation
I am already working on this, will update this description as I fix it. We do not scale the loss by number of batches to accumulate, so we end up with too large of gradients when doing gradient accumulation. 

More details here: https://github.com/fastai/fastai2/pull/194

### PR DESCRIPTION
**Pull request by marii-moe**
We were not dividing by the number of batches to accumulate, so this effectively increasing the learning rate. Added test to make sure this is fixed. I think this thread was lost when fastai/fastai2 got moved to fastai/fastai: https://github.com/fastai/fastai2/pull/194

fixes #3023

### CODE REVIEW

**Comment by jph00** [comment]
Nice one!

**Comment by muellerzr** [comment]
I know this is fixed and merged now, but I'm hesitant to use it. I was seeing better results with the old implementation of the high LR and worse results here. Do you have any advice @marii-moe as to how I should adapt my old LR to work with this new adjustment?

**Comment by marii-moe** [comment]
> 
> 
> I know this is fixed and merged now, but I'm hesitant to use it. I was seeing better results with the old implementation of the high LR and got worse results here. Do you have any advice @marii-moe as to how I should adapt my old LR to work with this new adjustment?

I am not familiar with your particular example, but you should get approximately the same results if you set your learning rate like so: 
new_lr = old_lr*n_acc/bs
